testsuite/tests/shape-index/index_constrs_records.ml

```diff
@@ -42,3 +42,10 @@ let e = Exn { l_exn = 2}
 let _ = match e with
   | Exn { l_exn } -> l_exn
   | _ -> assert false
+
+module Mod = struct
+  type t = A
+end
+let () =
+  match Mod.A with
+  | Mod.A -> ()
```

Comment:
I would suggest to add the record case since this is the case where the use of `Pextra_type` is more irregular.